Veterans - Eligibility Verifier + Eligibility Type

machikoyasuda opened the pull request:
closes #1427 
closes #1426

## What this PR does
- [x] Create `EligibilityVerifier` for MST Veterans
- [x] Create `EligibilityType`  for MST Veterans
- [x] Create Terraform values stubs for:  `"MST_VETERAN_GROUP_ID"`, `"MST_VETERAN_VERIFIER_NAME"`
- [x] Spec: Update `benefit-select.cy.js` spec from looking for 2 radio inputs, to 3 inputs. Update Cypress helper method so it correctly chooses Courtesy Card radio button.
- [x] Copy: Eligibility Index page and add VA.gov url as well (but not the new external link icon design).
- [x] Copy: Enrollment Success page - Same as Login.gov 
- [x] Copy: Eligibility Start page: 2 paragraphs (except for the required_items bullet list - that's not part of `EligibilityVerifier`) 
- [x] UI: Eligibility Index page - Make sure US Veteran option is the 2nd choice.
- [ ] #1426 

## What configuration has been done
- [x] Update KeyVault development for EligibilityVerifier: Create and set these key/values
- [x] Update KeyVault test for EligibilityVerifier: Create and set these key/values
- [x] Create Veteran group in LittlePay
- [x] Update KeyVault development for EligibilityType
- [x] Update KeyVault test for EligibilityVerifier

## Screenshots

<img width="1512" alt="image" src="https://github.com/cal-itp/benefits/assets/3673236/fbb24b9a-3703-4627-a127-faa9738fbbe0">
<img width="1512" alt="image" src="https://github.com/cal-itp/benefits/assets/3673236/fae6140f-b4f7-45e9-886c-20cbb21d6e6a">

## Documentation

### EligibilityVerifier model
- Adds `start_item_secondary_details`, an optional `TextField()`, to `EligibilityVerifier` model

### MediaItem viewmodel
- Adds `secondary_details`, an optional string, to `MediaItem` viewmodel class

The `secondary_details` is this details paragraph that comes after the bullets, on the Eligibility Start page.

<img width="1090" alt="image" src="https://github.com/cal-itp/benefits/assets/3673236/8b1ff283-4d68-4bb2-828b-47f443e1f8a9">


## Notes

On Production, after going through the Login.gov flow and verifying eligibility, the user sees the `enrollment` screen with this copy, consisting of 2 media items:
<img width="1512" alt="image" src="https://github.com/cal-itp/benefits/assets/3673236/4956f77e-14db-4a21-b200-f208c172a0c3">

In the new design, however, there is only 1 media item for all `enrollment` screens, and the remaining media item for Little Pay is also shortened from 2 paragraphs to 1:
<img width="688" alt="image" src="https://github.com/cal-itp/benefits/assets/3673236/39376ff2-36c7-47be-bb7f-04a92e635a98">

machikoyasuda commented:
@thekaveman @angela-tran I'm blocked here: https://github.com/cal-itp/benefits/issues/1437 Do you see any syntax error or something I'm doing wrong?

machikoyasuda commented on tests/cypress/plugins/helpers.js:
```diff
@@ -26,7 +26,7 @@ export const selectCourtesyCard = () => {
   cy.location("pathname").should("eq", `/eligibility/${agency.slug}`);
 
   // TODO find a more robust way to do this
```
This TODO apt is very appropriate 😅  Ideally the code would say something like: "Look for the Courtesy Card form input text", "Select the input closest to the Courtesy Card text"

machikoyasuda replied:
Should create a ticket for the cooldown time to refactor this method.

thekaveman replied:
#1443

machikoyasuda commented on terraform/app_service.tf:
```diff
@@ -96,6 +97,7 @@ resource "azurerm_linux_web_app" "main" {
     "AUTH_PROVIDER_SCOPE"                                  = "${local.secret_prefix}auth-provider-scope)"
     "AUTH_PROVIDER_CLAIM"                                  = "${local.secret_prefix}auth-provider-claim)"
     "MST_OAUTH_VERIFIER_NAME"                              = "${local.secret_prefix}mst-oauth-verifier-name)"
+    "MST_VETERAN_VERIFIER_NAME"                            = "${local.secret_prefix}mst-veteran-verifier-name)"
```
@thekaveman @angela-tran This is all I did in this `app_service.tf` file. Am I supposed to configure something manually in KeyVault after this PR is merged to define some test values.. or something?

angela-tran replied:
Yeah, you can configure the value in Key Vault even before this PR is merged. You'll create a new secret called `mst-veteran-verifier-name` with a value similar to the names for other EligibilityVerifiers.

machikoyasuda replied:
Added `mst-veteran-verifier-name` to Key Vault (Dev).

machikoyasuda commented on benefits/core/migrations/0002_data.py:
```diff
@@ -13,6 +13,11 @@ def load_data(app, *args, **kwargs):
     mst_senior_type = EligibilityType.objects.create(
         name="senior", label="Senior Discount (MST)", group_id=os.environ.get("MST_SENIOR_GROUP_ID", "group1")
     )
+    mst_veteran_type = EligibilityType.objects.create(
+        name="veteran",
+        label="Veteran Discount (MST)",
+        group_id=os.environ.get("MST_VETERAN_GROUP_ID", "group3"),
+    )
```
@angela-tran @thekaveman Is this all we need for #1426? Does this PR also close #1426 or is there something more that it entails? Wasn't sure.

angela-tran replied:
I think #1426 also includes creating the group in Littlepay and adding the group ID to Key Vault in Azure for dev and test.

thekaveman replied:
This change seems out of scope for this PR? Why not leave it all for the PR that closes #1426?

thekaveman replied:
@machikoyasuda it looks like you did fold #1426 into this one (creating the LP group, configuring it in KeyVault, linking to the EligibilityType)?

machikoyasuda replied:
I just wasn't really sure what to put here otherwise: https://github.com/cal-itp/benefits/pull/1436/files#r1238987028

thekaveman replied:
https://github.com/cal-itp/benefits/pull/1436#discussion_r1238989101

angela-tran commented on benefits/core/migrations/0002_data.py:
```diff
@@ -156,6 +161,26 @@ def load_data(app, *args, **kwargs):
+        start_item_details=_("eligibility.pages.start.veteran.start_item.details"),
+        start_item_secondary_details=_("eligibility.pages.start.veteran.start_item.secondary_details"),
+        unverified_title=_("eligibility.pages.unverified.login_gov.title"),
+        unverified_blurb=_("eligibility.pages.unverified.login_gov.p[0]"),
```
I wonder if we should rename this PO entry since it is now being used for the Login.gov `EligibilityVerifier` and the veteran `EligibilityVerifier`? 

Also, I'm looking at how we have a [different PO entry being used for Courtesy Cards](https://github.com/cal-itp/benefits/pull/1436/files#diff-f4aa1ce24b8c65ccf8b80dacab3cb3aff3e95d9bf430e8b22bc66eb233f40b23R215) ([`eligibility.pages.unverified.mst_cc.title`](https://github.com/cal-itp/benefits/blob/dev/benefits/locale/en/LC_MESSAGES/django.po#L121)), but... does the "Developer Benefits Full App" page in Figma imply now that all `unverified_title` and `unverified_blurb`s should be the same?

![image](https://github.com/cal-itp/benefits/assets/25497886/46fbe358-1690-40ea-8d9b-0291dd44aca3)

![image](https://github.com/cal-itp/benefits/assets/25497886/49f191d5-9aa7-497d-a468-78eaaad8d1a4)


cc @srhhnry @thekaveman

angela-tran replied:
Side thought -- this just gave me an idea for something to try during dev-design handoff (I worry I'll forget it if I wait until retro): 

As a part of design-dev handoff, could we go ahead and make GitHub tickets at that point to capture the changes that need to be done?

machikoyasuda replied:
Yeah and now the enrollment page copy is all the same - so there's some more copy string consolidation that can be done. But not sure what's in-scope vs. out-of-scope in this ticket and in this Sprint overall.

angela-tran commented on benefits/eligibility/views.py:
```diff
@@ -106,6 +106,7 @@ def start(request):
         icon=viewmodels.Icon("idcardcheck", pgettext("image alt text", "core.icons.idcardcheck")),
         heading=_(verifier.start_item_heading),
         details=_(verifier.start_item_details),
+        secondary_details=_(verifier.start_item_secondary_details),
```
Noticed a typo
```suggestion
        secondary_details=_(verifier.start_item.secondary_details),
```

angela-tran replied:
Oops, never mind! Got mixed up between the field and the PO msgid

angela-tran reviewed (commented):
These changes look good to me @machikoyasuda! I'm inclined to approve, but would like to get others' feedback about the PO entry naming / consolidation. Maybe at the very least we can remove `login_gov` from that PO entry msgid and leave the Courtesy Card verifier alone for now.

machikoyasuda commented on benefits/locale/en/LC_MESSAGES/django.po:
```diff
@@ -71,6 +71,36 @@ msgstr ""
+msgstr "Veteran benefit information"
+
+msgid "eligibility.pages.start.veteran.headline"
+msgstr "You selected a Veteran benefit."
```
should be `You selected a Veteran transit benefit.`

machikoyasuda commented:
@thekaveman @angela-tran Ready for re-review, hopefully final review. I added a few changes since the last review from Angela:

1. The copy on Eligibility Start was off by one word.
1. Removed `login_gov` from `msgid`s for Unverified page, b/c all the Unverified pages have the same copy now.
1. Added VA.gov link (which required adding `format_html()` code)

thekaveman reviewed (approved):
Please update the description to also note that this PR closes #1426 

Otherwise this looks good and can be approved once the Test KeyVault secrets are created!

machikoyasuda commented on benefits/core/migrations/0002_data.py:
```diff
@@ -156,6 +161,26 @@ def load_data(app, *args, **kwargs):
 
+    mst_veteran_verifier = EligibilityVerifier.objects.create(
+        name=os.environ.get("MST_VETERAN_VERIFIER_NAME", "VA.gov - Veteran (MST)"),
+        eligibility_type=mst_veteran_type,
```
@thekaveman Should I just put some other variable here then? Like just stub it with `mst_senior_type`?

thekaveman replied:
No, I think it's fine that this PR closes both (and I see the reasoning) -- just want to make sure it fully closes both (LP config, KeyVaults, etc.)

machikoyasuda commented:
@angela-tran No not yet. 

This is my current status:

<img width="840" alt="image" src="https://github.com/cal-itp/benefits/assets/3673236/dfad640b-31c8-4cd0-965f-63cb151701d8">

machikoyasuda commented:
- Created MST Veteran group on QA (Documented here https://github.com/cal-itp/benefits-secrets/pull/36)
- Updated KeyVault Dev with the new group id
- Updated KeyVault Test with the new group id and the verifier name

@angela-tran @thekaveman 

Ready for final...final 😅 ...review

angela-tran reviewed (approved):
Looks good to me @machikoyasuda 👍

thekaveman commented:
Ah @machikoyasuda I was still reviewing this!

thekaveman commented on benefits/locale/en/LC_MESSAGES/django.po:
```diff
@@ -5,8 +5,8 @@
 msgstr ""
-"Report-Msgid-Bugs-To: https://github.com/cal-itp/benefits/issues \n"
-"POT-Creation-Date: 2023-01-26 16:08+0000\n"
+"Report-Msgid-Bugs-To: \n"
```
Why did we lose our Issues link here?

thekaveman replied:
It looks like the helper `bin/makemessages.sh` was not run, it is supposed to put this line back.

machikoyasuda commented:
@thekaveman Oops sorry!!! 😓 

I just noticed I get this error after running `bin/makemessages.sh` now:

```
calitp@cdbe8faf8c05:~/app$ bin/makemessages.sh
processing locale es
processing locale en
sed: couldn't open temporary file benefits/locale/en/LC_MESSAGES/sed3E0pRF: Permission denied
```

which fails when running `sed -i 's/Report-Msgid-Bugs-To\:/Report-Msgid-Bugs-To\: https\:\/\/github.com\/cal-itp\/benefits\/issues/g' benefits/locale/*/*/*.po` - so that's why that line failed to get changed.

machikoyasuda commented:
Reported here: https://github.com/cal-itp/benefits/issues/1455
